controller/controller.go

```diff
@@ -192,11 +192,9 @@ func (c controller) GroupNeedsUpdate(req Request) (Request, bool, error) {
 
 func (c controller) Submit(ctx context.Context, req Request) (*task.Task, error) {
 	c.Config.Logger.Debug(ctx, "controller: handling submit")
-
 	if ok, err := ValidateSubmitRequest(req); !ok {
-		return nil, maskAny(err)
+		return nil, err
```

Comment:
Use `errgo.Cause()` then please. The `maskAny` adds a stack trace and we do this everywhere.